ethcore/src/block.rs

```diff
@@ -187,7 +187,7 @@ impl<'x> OpenBlock<'x> {
 
 	/// Push transactions onto the block.
 	#[cfg(not(feature = "slow-blocks"))]
-	fn push_transactions(&mut self, transactions: Vec<SignedTransaction>) -> Result<(), Error> {
+	fn push_transactions(&mut self, transactions: &[SignedTransaction]) -> Result<(), Error> {
```

Comment:
even if we end with going with clone all transactions approach, I would prefer to pass here a `Vec<SignedTransaction>` and make a clone higher up the stack, otherwise it goes against what this PR is trying to achieve
>  Another aspect of unnecessary cloning is that it obfuscates the intention of the code making it harder to read and follow